flask_hw/flask_rest/api/room/rooms.py

```diff
@@ -0,0 +1,140 @@
+import logging
+
+from flask_restful import Resource, marshal_with, marshal
+from sqlalchemy.exc import SQLAlchemyError
+
+from api.room.room_parsers import data_valid_for
+from api.room.structure import room_structure
+from db import Rooms, db
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+class RoomsRes(Resource):
+
+    @marshal_with(room_structure)
+    def get(self, number=None):
+        room_all = Rooms.query.all()
+        header = {
+            "Cache-Control": "no-store, no-cache, must-revalidate",
+            "Pragma": "no-cache"
+        }
+        args = data_valid_for('GET')
+        logging.debug(room_all)
+        logging.debug(number)
+        logging.debug(args)
+        if number:
+            room = Rooms.query.get(number)
+            if room:
+                return room, header
+            else:
+                # FIXME: how to display message instead of Rooms
+                #  structure with nulls?
+                return {"message": "room not found"}, 404, header
+        else:
+            if args['status']:
+                rooms = Rooms.query.filter_by(status=args['status']).all()
+                # todo: add if status not found (or empty list is OK?)
+                if not rooms:
+                    msg = f"rooms with status='{args['status']}' not found"
+                    return {"message": msg}, 404, header
+                return rooms, header
+        return room_all, header
+
+    def post(self):
+        data = data_valid_for('POST')
+        logging.debug(data)
+        if not Rooms.query.get(data['number']):
+            new_room = Rooms(**data)
+            db.session.add(new_room)
+            # FIXME: what if room was not saved?
+            try:
+                db.session.commit()
+            except SQLAlchemyError:
+                return {"message": "could not save to database, try later"}
+            location = {
+                "Location": f'/api/v0.1/rooms/{str(new_room.number)}'
```

Comment:
do you really need location?

Reply:
If successful, the server should return 201 Created with an empty body, but with an additional header
`Location: /: entity /: new_id`
indicating the location of the created object.
Most often, it is not required to return the response body, since the client has all the necessary data, and he can get the` id` of the created object from Location.